metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/dao/DatasetDao.java

```diff
@@ -74,9 +74,10 @@ public DatasetDao(MorphiaDatastoreProvider morphiaDatastoreProvider,
    */
   @Override
   public String create(Dataset dataset) {
-    Dataset datasetSaved = ExternalRequestUtil
-        .retryableExternalRequestForNetworkExceptions(
-            () -> morphiaDatastoreProvider.getDatastore().save(dataset));
+    final ObjectId objectId = Optional.ofNullable(dataset.getId()).orElseGet(ObjectId::new);
+    dataset.setId(objectId);
```

Comment:
Yes there is a good reason for this.
To make our retries idempotent otherwise they can fail on subsequent calls, maybe we haven't experienced it but from what I saw it could happen..
This should prevent that

Reply:
I think I understand this. I'm not sure how this can be an issue in practice, but I'm ok changing this. I do believe though that this will be annoying to remember to do everywhere in the code. And doesn't Morphia do this internally?

Reply:
Mongo java driver does have a secure retry on network failure but only once.
Perhaps we could maybe get away with that but I'm completely sure.
I feel this way we are way more secure.